From 3c868f548cc8d201e1663bc69315b22d3112d7fe Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Mon, 19 Nov 2018 21:57:51 +0100
Subject: [PATCH] makepkg: add /dev/pts mount option

 some sw like julia and probably others want /dev/pts mounted into chroot
 add -P command line option which mounts /dev/pts for such an app.
 NOTE: there is no 'options=()' for that yet , not even sure we want that
---
 scripts/makepkg | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index 18ee3575..84a616dd 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -48,6 +48,7 @@ INSTALL=0
 DOWNLOAD=""
 NOBUILD=0
 NOCCACHE=0
+PTS=0
 NODEPS=0
 NOEXTRACT=0
 NOSTRIP=0
@@ -356,6 +357,7 @@ usage() {
 	$ECHO "  -n, --nostrip    Do not strip binaries/libraries"
 	$ECHO "  -o, --nobuild    Download and extract files only"
 	$ECHO "  -p <buildscript> Use an alternate build script (instead of $BUILDSCRIPT)"
+	$ECHO "  -P, --pts        Mount /dev/pts"
 	$ECHO "  -r, --rmdeps     Remove installed dependencies after a successful build"
 	$ECHO "  -R, --chroot     Build the package in a chroot environment (default)"
 	$ECHO "  -H, --host       Build the package on the host system"
@@ -381,6 +383,9 @@ usage() {
 chroot_umount() {
     msg "Attempting to umount chroot directories..."
     umount $CHROOTDIR/dev/shm >/dev/null
+    if [ "$PTS" = "1" ]; then
+	umount $CHROOTDIR/dev/pts >/dev/null
+    fi
     umount $CHROOTDIR/proc >/dev/null
     umount $CHROOTDIR/sys >/dev/null
     umount $CHROOTDIR/dev >/dev/null
@@ -401,6 +406,9 @@ chroot_mount() {
     mount -t sysfs none $CHROOTDIR/sys >/dev/null
     mount -t devtmpfs none $CHROOTDIR/dev >/dev/null
     mount -o bind /dev/shm $CHROOTDIR/dev/shm
+    if [ "$PTS" = "1" ]; then
+	mount -o bind /dev/pts $CHROOTDIR/dev/pts >/dev/null
+    fi
     mount -o bind /var/cache/pacman-g2 $CHROOTDIR/var/cache/pacman-g2 >/dev/null
     if [ "$NOCCACHE" = "0" -a -d /usr/lib/ccache/bin -a ! "`check_option NOCCACHE`" ]; then
 	mount -o bind $CCACHE_BASEDIR/$pkgname \
@@ -703,6 +711,7 @@ while [ "$#" -ne "0" ]; do
 		--sudosync)   DEP_SUDO=1 ;;
 		--builddeps)  DEP_SRC=1 ;;
 		--noccache)   NOCCACHE=1 ;;
+		--pts)        PTS=1;;
 		--nodeps)     NODEPS=1 ;;
 		--noextract)  NOEXTRACT=1 ;;
 		--install)    INSTALL=1 ;;
@@ -726,7 +735,7 @@ while [ "$#" -ne "0" ]; do
 			exit 1
 			;;
 		-*)
-			while getopts "aA:bBcCdD:efgGhij:kl:Lmnop:urRHsSt:w:z:-" opt; do
+			while getopts "aA:bBcCdD:efgGhij:kl:LmnoPp:urRHsSt:w:z:-" opt; do
 				case $opt in
 					a) SEARCHDEPS=1 ;;
 					A) CACHEURL=$OPTARG ;;
@@ -750,6 +759,7 @@ while [ "$#" -ne "0" ]; do
 					m) USE_COLOR="n" ;;
 					n) NOSTRIP=1 ;;
 					o) NOBUILD=1; CHROOT=0 ;;
+					P) PTS=1;;
 					p) BUILDSCRIPT=$OPTARG ;;
 					r) RMDEPS=1 ;;
 					R) CHROOT=1 ;;
-- 
2.19.1

